spring-integration-zip/src/main/java/org/springframework/integration/zip/splitter/UnZipResultSplitter.java

```diff
@@ -43,7 +48,9 @@
 			final String filename = FilenameUtils.getName(entry.getKey());
 			final Message<Object> splitMessage = MessageBuilder.withPayload(entry.getValue())
 					.setHeader(FileHeaders.FILENAME, filename)
-					.setHeader(ZipHeaders.ZIP_ENTRY_PATH, path).build();
+					.setHeader(ZipHeaders.ZIP_ENTRY_PATH, path)
+					.copyHeaders(headers)
```

Comment:
I think in this context the `copyHeadersIfAbsent()` would be better.
In case of an upstream message already has the mentioned `FileHeaders.FILENAME` or `ZipHeaders.ZIP_ENTRY_PATH` headers.
If we use this `copyHeaders()`, we override just added headers based on the zip entry. And that is wrong.
I think that should be covered with the test-case as well.